platforms/ide/tooling-api/src/main/java/org/gradle/tooling/internal/consumer/parameters/BuildProgressListenerAdapter.java

```diff
@@ -1186,15 +1186,18 @@ private static Failure toFailure(InternalFailure origFailure) {
         if (origFailure == null) {
             return null;
         }
-        List<InternalBasicProblemDetailsVersion3> problems = new ArrayList<>();
+        List<InternalBasicProblemDetailsVersion3> problemDetails = new ArrayList<>();
         try {
-            problems.addAll(origFailure.getProblems());
+            problemDetails.addAll(origFailure.getProblems());
         } catch (AbstractMethodError ignore) {
             // Older Gradle versions don't have this method
         }
-        List<Problem> clientProblems = new ArrayList<>(problems.size());
-        for (InternalBasicProblemDetailsVersion3 problem : problems) {
-            clientProblems.add(createProblemReport(problem));
+        List<Problem> clientProblems = new ArrayList<>(problemDetails.size());
+        for (InternalBasicProblemDetailsVersion3 problemDetail : problemDetails) {
+            if (problemDetail == null) { // Should not happen, but with some older snapshot versions we see this.
+                continue;
+            }
+            clientProblems.add(createProblemReport(problemDetail));
```

Comment:
🤔 Maybe this is a bit less terse?
```suggestion
            if (problemDetail != null) { 
                // Details normally shouldn't be null, but with some older snapshots, it can happen.
                 clientProblems.add(createProblemReport(problemDetail));
            }
```